github/repos_releases.go

```diff
@@ -304,6 +310,25 @@ func (s *RepositoriesService) DownloadReleaseAsset(ctx context.Context, owner, r
 	return resp.Body, "", nil
 }
 
+func (s *RepositoriesService) downloadReleaseAssetFromURL(ctx context.Context, url string) (rc io.ReadCloser, err error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req = withContext(ctx, req)
+	req.Header.Set("Accept", "*/*")
+	httpClient := &http.Client{}
```

Comment:
Thank you, @andersjanmyr - this is why we didn't do this originally... because this will not work for all users of this client library (for the reasons I explained earlier).

I like this solution... though I wondering if we could perform a simple change to make it work for everybody?

I'm thinking that... instead of passing `followRedirects bool`, what if we were to pass: `followRedirectsClient *http.Client`?

If `followRedirectsClient` is nil, no redirect following is attempted and the `redirectURL` string is returned as before.

If, however, `followRedirectsClient` is non-nil, it will call your new function and pass in the client!

Now, people will be able to build whatever wacky `http.Client` they need, pass it in, and let your code do its job.
If they don't need anything special, they could just pass in `http.DefaultClient`.

In fact, the documentation should state this particular case as an example for the majority of the users.
However, power users should be told that they can pass in any kind of `http.Client` that they want if they wish redirects to be followed.

Does that make sense?  What do you think about making this change?

Reply:
Okay, it sounds reasonable. I'll make the change

Reply:
Done.